File: cloud-collect-core/src/main/java/com/windea/demo/cloudcollect/core/service/CollectCategoryService.java
package com.windea.demo.cloudcollect.core.service;

import com.windea.demo.cloudcollect.core.domain.entity.*;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

/**
 * 收藏的分类的服务。
 */
public interface CollectCategoryService {
	/**
	 * 创建自己的分类。
	 */
	void create(CollectCategory category, User user);

	/**
	 * 删除自己的分类。
	 */
	void delete(Long id);

	/**
	 * 修改自己的分类。名字，概述。
	 */
	void modify(Long id, CollectCategory category);

	/**
	 * 得到某一分类。
	 */
	CollectCategory get(Long id);

	/**
	 * 分页得到某一分类的所有收藏。
	 */
	Page<Collect> getCollectPage(Long id, Pageable pageable);

	/**
	 * 得到某一分类的收藏数量。
	 */
	Long getCollectCount(Long id);

	/**
	 * 分页得到所有分类。
	 */
	Page<CollectCategory> findAll(Pageable pageable);

	/**
	 * 分页得到某一用户的所有分类。
	 */
	Page<CollectCategory> findByUser(Long userId, Pageable pageable);

	/**
	 * 根据名字分页模糊查询某一用户的所有分类。
	 */
	Page<CollectCategory> findByUserAndName(Long userId, String name, Pageable pageable);

	/**
	 * 检查某一分类是否已存在。
	 */
	boolean exists(CollectCategory category);
}

File: cloud-collect-core/src/main/kotlin/com/windea/demo/cloudcollect/core/service/CollectCategoryService.kt
package com.windea.demo.cloudcollect.core.service

import com.windea.demo.cloudcollect.core.domain.entity.*
import org.springframework.data.domain.*

interface CollectCategoryService {
	/**创建自己的分类。*/
	fun create(category: CollectCategory, user: User)
	
	/**修改自己的分类。*/
	fun modify(category: CollectCategory)
	
	/**删除自己的分类。*/
	fun deleteById(id: Long)
	
	/**根据id得到某一分类。*/
	fun findById(id: Long): CollectCategory
	
	/**得到所有分类。*/
	fun findAll(pageable: Pageable): Page<CollectCategory>
	
	/**根据名字和用户id模糊查询所有分类。*/
	fun findAllByNameContainsAndUserId(userId: Long, name: String, pageable: Pageable): Page<CollectCategory>
	
	/**根据用户id查询所有分类。*/
	fun findAllByUserId(userId: Long, pageable: Pageable): Page<CollectCategory>
	
	/**检查某一分类是否已存在。*/
	fun existsByNameAndUserId(name: String, userId: Long): Boolean
	
	
	/**得到该分类的所有收藏。*/
	fun getCollectPage(id: Long, pageable: Pageable): Page<Collect>
}
